kura/org.eclipse.kura.nm/src/main/java/org/eclipse/kura/nm/NMVersion.java

```diff
@@ -0,0 +1,111 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Areti and others
+ * 
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ * 
+ * Contributors:
+ *  Areti
+ *******************************************************************************/
+package org.eclipse.kura.nm;
+
+import java.util.Objects;
+
+public class NMVersion implements Comparable<NMVersion> {
```

Comment:
Didn't notice its existance :)

After a few tests I've found that is breaks on non numeric characters and we'll end up using `.compareTo(Version.parseVersion(x)>=0` for each comparison or wrapping it with the additional logic needed 🤔

Reply:
🤔 I don't expect to find alphanumeric characters for NM version, and yeah it's not the best from a readability point of view... but it's something we already use in the codebase and doesn't need additional unit tests.

🤷‍♂️ Up to you... :)

Reply:
I think, at this point, you want to keep this class (which is ok for me). But `NMVersion` doesn't really make much sense since it just represents a Semantic Version not the version for NetworkManager specifically. We could use the same class for representing the ModemManager or WPA Supplicant versions... could that be a better name? `SemanticVersion`?

Reply:
Ended up finding a `1.30.0-1` as a NetworkManager version, which would not work on `org.osgi.framework.Version`. Not sure if it's creativity from the OEM or something we can realistically encounter out in the field, but it's still breaking for my use case 👼 
OK on renaming it 👍